test/js-api/jsapi.js

```diff
@@ -661,6 +664,112 @@ test(() => {
+    assert_equals(typeof globalDesc.value, "function");
+    assert_equals(globalDesc.writable, true);
+    assert_equals(globalDesc.enumerable, false);
+    assert_equals(globalDesc.configurable, true);
+    Global = WebAssembly.Global;
+}, "'WebAssembly.Global' data property");
+
+test(() => {
+    const globalDesc = Object.getOwnPropertyDescriptor(WebAssembly, 'Global');
+    assert_equals(Global, globalDesc.value);
+    assert_equals(Global.length, 1);
+    assert_equals(Global.name, "Global");
+    assertThrows(() => Global(), TypeError);
+    assertThrows(() => new Global(1), TypeError);
+    assertThrows(() => new Global({}), TypeError);
+    assertThrows(() => new Global({type: 'foo'}), TypeError);
+    assertThrows(() => new Global({type: 'i64'}), TypeError);
+    assert_equals(new Global({type:'i32'}) instanceof Global, true);
+    assert_equals(new Global({type:'f32'}) instanceof Global, true);
+    assert_equals(new Global({type:'f64'}) instanceof Global, true);
+    assert_equals(new Global({type:'i32', mutable: false}) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: false}) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: false}) instanceof Global, true);
+    assert_equals(new Global({type:'i32', mutable: true}) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: true}) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: true}) instanceof Global, true);
+    assert_equals(new Global({type:'i32'}, 0x132) instanceof Global, true);
+    assert_equals(new Global({type:'f32'}, 0xf32) instanceof Global, true);
+    assert_equals(new Global({type:'f64'}, 0xf64) instanceof Global, true);
+    assert_equals(new Global({type:'i32', mutable: false}, 0x132) instanceof Global, true);
+    assert_equals(new Global({type:'f32', mutable: false}, 0xf32) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: false}, 0xf64) instanceof Global, true);
+    assert_equals(new Global({type:'i32', mutable: true}, 0x132) instanceof Global, true);
+    assert_equals(new Global({type:'f32', mutable: true}, 0xf32) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: true}, 0xf64) instanceof Global, true);
+}, "'WebAssembly.Global' constructor function");
+
+test(() => {
+    const globalProtoDesc = Object.getOwnPropertyDescriptor(Global, 'prototype');
+    assert_equals(typeof globalProtoDesc.value, "object");
+    assert_equals(globalProtoDesc.writable, false);
+    assert_equals(globalProtoDesc.enumerable, false);
+    assert_equals(globalProtoDesc.configurable, false);
+}, "'WebAssembly.Global.prototype' data property");
+
+test(() => {
+    const globalProtoDesc = Object.getOwnPropertyDescriptor(Global, 'prototype');
+    globalProto = Global.prototype;
+    assert_equals(globalProto, globalProtoDesc.value);
+    assert_equals(String(globalProto), "[object WebAssembly.Global]");
```

Comment:
`String(globalProto)` should throw because it should call `@@toPrimitiveMethod` where this is not a WebAssembly.Global object.
It should be `globalProto.toString()` or `Object.prototype.toString.call(globalProto)`